qiime2/sdk/plugin_manager.py

```diff
@@ -81,15 +81,25 @@ def _init(self, add_plugins):
                 package = entry_point.module_name.split('.')[0]
                 plugin = entry_point.load()
 
-                self.add_plugin(plugin, package, project_name)
+                self.add_plugin(plugin, package, project_name,
+                                consistency_check=False)
 
-    def add_plugin(self, plugin, package=None, project_name=None):
+            self.add_plugin(plugin, package, project_name,
+                            consistency_check=True)
```

Comment:
I would call `self._consistency_check()` here instead, so that the plugin isn't added twice.